### PR TITLE
New publisher extensions

### PR DESCRIPTION
**Pull request by slopes321**
Adding support for the following publishers:
- brakeman
- claim_broken_build
- cobertura_report
- email_ext
- html_publisher
- publish_tap_results
- xunit

### CODE REVIEW

File: lib/jenkins_pipeline_builder/extensions/publishers.rb [hunk 1]
```diff
@@ -298,3 +298,424 @@
...
+  xml do |allow_claim|
+    send('hudson.plugins.claim.ClaimPublisher', 'plugin' => 'claim') if allow_claim
+  end
+end
+
+publisher do
+  name :cobertura_report
+  plugin_id 'cobertura'
+  description 'This plugin integrates Cobertura coverage reports to Jenkins.'
+  jenkins_name 'This plugin integrates Cobertura coverage reports to Jenkins.'
```
Comment (crimsonknave) [inline lib/jenkins_pipeline_builder/extensions/publishers.rb]: Is this actually what the label for it is in jenkins?

**Comment by crimsonknave** [comment]
Thanks for this! That is a lot of plugins!

File: lib/jenkins_pipeline_builder/extensions/publishers.rb [hunk 2]
```diff
@@ -298,3 +298,424 @@
...
+
+      def send_metric_targets(target, thresholds)
+        case target
+        when :healthy
+          name = 'healthyTarget'
+        when :unhealthy
+          name = 'unhealthyTarget'
+        when :failing
+          name = 'failingTarget'
+        end
```
Comment (crimsonknave) [inline lib/jenkins_pipeline_builder/extensions/publishers.rb]: Can this just be collapsed into name = "#{target}Target" ?

File: lib/jenkins_pipeline_builder/extensions/publishers.rb [hunk 3]
```diff
@@ -298,3 +298,424 @@
...
+    end
+  end
+
+end
+
+publisher do
+  name :xunit
+  plugin_id 'xunit'
+  description 'This plugin makes it possible to record xUnit test reports.'
+  jenkins_name 'This plugin makes it possible to record xUnit test reports.'
```
Comment (crimsonknave) [inline lib/jenkins_pipeline_builder/extensions/publishers.rb]: Is this actually the name in the jenkins UI?

**Comment by crimsonknave** [comment]
So, there are a couple of places with some large helper methods that I'd like to pull out of the xml. Ideally these would go into a PublisherHelpers class that gets included in the publishers.rb file. Specifically, I'd like to avoid having methods inside the xml block if possible and any setting of defaults I think could be done in there.

At the moment I'm not sure if I prefer defining a helper class for each specific thing and then you'd call `PublisherHelpers::Cobetura.defaults` or just having a single helper that is included at the top of the file and then the helper methods would be namespaced by the plugin: `cobetura_defaults`.

File: lib/jenkins_pipeline_builder/extensions/publishers.rb (hunk 1 above)
Comment (slopes321) [inline lib/jenkins_pipeline_builder/extensions/publishers.rb]: Ah, no, it looks I using jenkins_name incorrectly when I had these as standalone extensions in my project. I'll update them.

File: lib/jenkins_pipeline_builder/extensions/publishers.rb (hunk 2 above)
Comment (slopes321) [inline lib/jenkins_pipeline_builder/extensions/publishers.rb]: Now that you mentioned it ... yes. :)

File: lib/jenkins_pipeline_builder/extensions/publishers.rb (hunk 3 above)
Comment (slopes321) [inline lib/jenkins_pipeline_builder/extensions/publishers.rb]: I'll update this one too.

File: lib/jenkins_pipeline_builder/extensions/publishers.rb [hunk 4]
```diff
@@ -298,3 +298,417 @@
...
+  plugin_id 'cobertura'
+  description 'This plugin integrates Cobertura coverage reports to Jenkins.'
+  jenkins_name 'Cobertura Plugin'
+  announced false
+
+  xml do |params|
+    send('hudson.plugins.cobertura.CoberturaPublisher', 'plugin' => 'cobertura') do
+
+      def send_metric_targets(target, thresholds)
+        name = '#{target}Target'
```
Comment (crimsonknave) [inline lib/jenkins_pipeline_builder/extensions/publishers.rb]: You need double quotes here to do the string interpolation

**Pull request merged**